zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -202,11 +201,26 @@ public Interpreter getCurrentRepl() {
     return getRepl(getRequiredReplName());
   }
 
+  public List<InterpreterCompletion> getInterpreterCompletion() {
+    List<InterpreterCompletion> completion = new LinkedList();
+    for (InterpreterSetting intp: factory.getInterpreterSettings(note.getId())){
+      completion.add(new InterpreterCompletion(intp.getGroup(), intp.getGroup()));
+    }
+    return completion;
+  }
+
   public List<InterpreterCompletion> completion(String buffer, int cursor) {
+    String lines[] = buffer.split(System.getProperty("line.separator"));
+    if (lines.length > 0
+      && (lines[0].startsWith("%") && cursor <= lines[0].trim().length())) {
```

Comment:
Code text can follow after interpreter selection without line.separator. For example

```sql
%sql select * from table
```
In this case, `getInterpreterCompletion()` will be invoked even if cursor is in somewhere `select * from table`. isn't it?